app/upgrades/v14_2/upgrades.go

```diff
@@ -33,8 +36,30 @@ func CreateUpgradeHandler(
 			}
 		}
 
+		// Add EIP contained in Shanghai hard fork to the extra EIPs
+		// in the EVM parameters. This enables using the PUSH0 opcode and
+		// thus supports Solidity v0.8.20.
+		logger.Info("adding EIP 3855 to EVM parameters")
+		err := EnableEIPs(ctx, ek, 3855)
+		if err != nil {
+			logger.Error("error while enabling EIPs", "error", err)
+		}
+
 		// Leave modules are as-is to avoid running InitGenesis.
 		logger.Debug("running module migrations ...")
 		return mm.RunMigrations(ctx, configurator, vm)
 	}
 }
+
+// EnableEIPs enables the given EIPs in the EVM parameters.
+func EnableEIPs(ctx sdk.Context, ek *evmkeeper.Keeper, eips ...int64) error {
+	evmParams := ek.GetParams(ctx)
+
+	for _, eip := range eips {
+		if !slices.Contains(evmParams.ExtraEIPs, eip) {
```

Comment:
This should also be in the `params.Validate()` function

Reply:
done 👍